xarray/backends/api.py

```diff
@@ -1926,7 +1926,10 @@ def to_netcdf(
         if engine is None:
             engine = _get_default_engine(path_or_file)
         path_or_file = _normalize_path(path_or_file)
-    else:  # file-like object
+    elif engine not in ("scipy", "h5netcdf"):
+        # Requested engine is not compatible with writing to a file-like object.
+        # Ideally we would raise an error, but for backwards-compatibility with
+        # old behaviour we overwrite the engine to scipy.
         engine = "scipy"
```

Comment:
Can we issue a warning here? Also let's keep the comment (and maybe part of the existing code structure) to clarify what this case exists for.
```python
else:  # file-like object
    if engine not in ("scipy", "h5netcdf"):
        emit_user_level_warning(
            f"Requested {engine=} is not compatible with writing to a file-like object. "
            "This will raise an error in the future, for now defaulting to engine='scipy'.",
            FutureWarning,
        )
        engine = "scipy"
```

Reply:
Re code structure, I left the `else` and the `# file-like object` comment in place but it looks like the CI auto-fixes formatting commit got rid of it. Not sure how to disable that.